### PR TITLE
feat(edit-page-v2): Enable Navigation Sorting feature

### LINKED ISSUE #28040: Edit Page V2: Enable Navigation Sorting Feature
### Parent
#27546 

### Task description
As a dotCMS user, I want to be able to sort the navigation on my website so that I can easily organize and rearrange the menu items based on my preferences.

Let's create a new macro with HTML and CSS inline so we can control the styling, use this one as base:

```
#macro(orderMenu)
    #if ($EDIT_MODE && $PUBLISH_HTMLPAGE_PERMISSION)
        #set($menuLevel = $VTLSERVLET_URI.split('/').size() - 1)

        <a href="#" onclick="openReorderDialog(event)" class="btn btn-primary btn-xs normaltip" data-original-title="Reorder Menu">
            <i class="fa fa-arrow-up"></i>
            <i class="fa fa-arrow-down"></i>
        </a>
        <script>
            function openReorderDialog(event) {
                event.preventDefault();
                var reorderUrl = "${directorURL}&startLevel=${menuLevel}&depth=2&pagePath=${VTLSERVLET_URI}&hostId=${host.identifier}";
                var customEvent = window.top.document.createEvent("CustomEvent");
                customEvent.initCustomEvent("ng-event", false, false,  {
                    name: "reorder-menu",
                    data: reorderUrl
                });
                window.top.document.dispatchEvent(customEvent);
            }
        </script>
    #end
#end
```

Add the macro to the starter and make it work with the custom event.

```[tasklist]
### Acceptance Criteria
- [ ] User should be able to sort their nav in VTL pages
```

### External Links
N/A

### Assumptions & Initiation Needs
N/A

### PR DESCRIPTION
**Pull request by zJaaal**
### Proposed Changes
* Create new JSP Macro `sortNavigation()` to work with the new editor.
* Connect the Navigation Reorder Dialog with the new editor

### Additional Info
We need to replace the piece of code we add to add this button in the VTLs files with the new macro.

Ex:  In the current starter, we add this button in this file `application/themes/travel/header.vtl`
### Current Code to add the button
```
  ## ADDS ICON IN EDIT MODE TO REORDER SIDE NAVIGATION ##
            #if ($EDIT_MODE && $PUBLISH_HTMLPAGE_PERMISSION)
                #set($_formId="dot_form_menu_$date.date.time")
                <form class="dotcms__navbar-form" action="${directorURL}" method="post" name="${_formId}" id="${_formId}">
                    <input type="hidden" name="cmd" value="orderMenu" />
                    <input type="hidden" name="hostId" value="${host.identifier}" />
                    <input type="hidden" name="referer" value="${VTLSERVLET_URI}" />
                    <input type="hidden" name="pagePath" value="${VTLSERVLET_URI}" />
                    <input type="hidden" name="startLevel" value="1" />
                    <input type="hidden" name="depth" value="2" />
                    <a href="javascript:document.getElementById('${_formId}').submit();" class="reorder-menu-link" data-original-title="Reorder Menu">
                        <i class="arrow-up"></i> 
                        <i class="arrow-down"></i> 
                    </a>
                </form>
            #end
```

### Should be updated to
```
            ## ADDS ICON IN EDIT MODE TO REORDER SIDE NAVIGATION ##
            #if ($EDIT_MODE && $PUBLISH_HTMLPAGE_PERMISSION)
              #set($_formId="dot_form_menu_$date.date.time")
              #sortNavigation()
            #end
```

### Screenshots

https://github.com/dotCMS/core/assets/63567962/44a5f849-77e4-4dee-817f-3d19933b1740

### CODE REVIEW

**Comment by zJaaal** [comment]
Code is ready but I'm writting test cases. Leaving in draft meanwhile.

**Review by hmoreras — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: dotCMS/src/main/webapp/WEB-INF/velocity/VM_global_library.vm [hunk 1]
```diff
@@ -189,3 +189,64 @@
         </script>
     #end
 #end
+
+#macro(newOrderMenu)
```
Comment (fmontes) [inline dotCMS/src/main/webapp/WEB-INF/velocity/VM_global_library.vm:193]:
```suggestion
#macro(sortNavigation)
```

File: core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.spec.ts [hunk 2]
```diff
@@ -325,5 +325,16 @@ describe('DotEmaDialogComponent', () => {
...
+        it("should trigger openDialogOnURL in the store when it's a URL", () => {
+            const openDialogOnURLSpy = jest.spyOn(storeSpy, 'openDialogOnURL');
+
+            component.openDialogOnUrl('https://demo.dotcms.com/jsp.jsp', 'test');
+
+            expect(openDialogOnURLSpy).toHaveBeenCalledWith({
+                title: 'test',
+                url: 'https://demo.dotcms.com/jsp.jsp'
+            });
+        });
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.spec.ts:338]: Let's never do this, when we need to call a method in a component, let's use a service, when I did the refactor of the toolbar this was the only pain point.

**Review by fmontes — CHANGES_REQUESTED** [review]
We need to create a task to update the starter.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.spec.ts (hunk 2 above)
Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.spec.ts:338]: Noted, will consider this in the future.

**Review by fmontes — APPROVED** [review]

**Pull request merged**